src/coordinator/tsdb/remote/compressed_codecs.go

```diff
@@ -142,25 +158,52 @@ func CompressedSeriesFromSeriesIterator(it encoding.SeriesIterator) (*rpc.Series
 		})
 	}
 
-	tags, err := compressedTagsFromTagIterator(it.Tags())
-	if err != nil {
-		return nil, err
-	}
-
 	start := xtime.ToNanoseconds(it.Start())
 	end := xtime.ToNanoseconds(it.End())
+	tagIter := it.Tags()
+	compressedTags, canCompress := compressedTagsFromTagIteratorWithEncoder(tagIter, iterPools)
+
+	var tags []*rpc.Tag
+	var err error
+	if !canCompress {
```

Comment:
maybe not rely on an error to switch logic ? Ideally, if we have compressedDatapoints we should always have compressedTags ? Otherwise, its confusing.

Reply:
This is going the other way; if we *can* make compressedTags, we do, otherwise we use regular tags